### ISSUE TITLE
Stuck OSD prepare job prevents operator spin up

### ISSUE THREAD

**Issue by stephan2012** [rook/rook#4495]
**Is this a bug report or feature request?**
* Bug Report

**Deviation from expected behavior:**
When a OSD prepare job is stuck for any reason the operator does not continue to spin up. For example, it does not process cephblockpool CRDs. In our case, there was a problem with the prepare job for one node that hung more many weeks for an unknown reason.

**Expected behavior:**
Operator should treat hanging prepare jobs appropriately. E.g., very old jobs should be deleted and recreated. Jobs that do not complete within a sensible timeout should be retried later without blocking the operator.

**How to reproduce it (minimal and precise):**

We experienced this behavior after upgrading from 1.1.2 to 1.1.7.

**File(s) to submit**:

Operator logs:

```
2019-12-12 13:52:40.903558 I | op-osd: start provisioning the osds on nodes, if needed
2019-12-12 13:52:40.907629 I | op-osd: 4 of the 4 storage nodes are valid
2019-12-12 13:52:40.913064 I | op-k8sutil: Found previous job rook-ceph-osd-prepare-worker-01. Status={Conditions:[{Type:Complete Status:True LastProbeTime:2019-10-22 13:27:42 +0000 UTC LastTransitionTime:2019-10-22 13:27:42 +0000 UTC Reason: Message:}] StartTime:2019-10-22 13:25:33 +0000 UTC CompletionTime:2019-10-22 13:27:42 +0000 UTC Active:1 Succeeded:1 Failed:0}
2019-12-12 13:52:40.913074 I | op-osd: osd provision job started for node worker-01
2019-12-12 13:52:40.921198 I | op-k8sutil: Removing previous job rook-ceph-osd-prepare-worker-02 to start a new one
2019-12-12 13:52:40.926827 I | op-k8sutil: batch job rook-ceph-osd-prepare-worker-02 still exists
2019-12-12 13:52:42.928968 I | op-k8sutil: batch job rook-ceph-osd-prepare-worker-02 deleted
2019-12-12 13:52:42.933079 I | op-osd: osd provision job started for node worker-02
2019-12-12 13:52:42.952852 I | op-k8sutil: Removing previous job rook-ceph-osd-prepare-worker-03 to start a new one
2019-12-12 13:52:42.965669 I | op-k8sutil: batch job rook-ceph-osd-prepare-worker-03 still exists
2019-12-12 13:52:44.968221 I | op-k8sutil: batch job rook-ceph-osd-prepare-worker-03 deleted
2019-12-12 13:52:44.973335 I | op-osd: osd provision job started for node worker-03
2019-12-12 13:52:44.992378 I | op-k8sutil: Removing previous job rook-ceph-osd-prepare-worker-04 to start a new one
2019-12-12 13:52:45.004794 I | op-k8sutil: batch job rook-ceph-osd-prepare-worker-04 still exists
2019-12-12 13:52:47.008039 I | op-k8sutil: batch job rook-ceph-osd-prepare-worker-04 deleted
2019-12-12 13:52:47.012740 I | op-osd: osd provision job started for node worker-04
2019-12-12 13:52:47.012750 I | op-osd: start osds after provisioning is completed, if needed
2019-12-12 13:52:47.017838 I | op-osd: osd orchestration status for node worker-01 is starting
2019-12-12 13:52:47.017869 I | op-osd: osd orchestration status for node worker-02 is completed
2019-12-12 13:52:47.017874 I | op-osd: starting 1 osd daemons on node worker-02
2019-12-12 13:52:47.031562 I | op-osd: deployment for osd 2 already exists. updating if needed
2019-12-12 13:52:47.035582 I | op-mon: this is not an upgrade, not performing upgrade checks
2019-12-12 13:52:47.035593 I | op-k8sutil: updating deployment rook-ceph-osd-2
2019-12-12 13:53:05.086256 I | op-k8sutil: finished waiting for updated deployment rook-ceph-osd-2
2019-12-12 13:53:05.086279 I | op-mon: this is not an upgrade, not performing upgrade checks
2019-12-12 13:53:05.086283 I | op-osd: started deployment for osd 2 (dir=false, type=bluestore)
2019-12-12 13:53:05.092803 I | op-osd: osd orchestration status for node worker-03 is orchestrating
2019-12-12 13:53:05.092820 I | op-osd: osd orchestration status for node worker-04 is starting
2019-12-12 13:53:05.092824 I | op-osd: 1/4 node(s) completed osd provisioning, resource version 32132516
2019-12-12 13:53:05.093964 I | op-osd: osd orchestration status for node worker-03 is completed
2019-12-12 13:53:05.093976 I | op-osd: starting 1 osd daemons on node worker-03
2019-12-12 13:53:05.101051 I | op-osd: deployment for osd 1 already exists. updating if needed
2019-12-12 13:53:05.102913 I | op-mon: this is not an upgrade, not performing upgrade checks
2019-12-12 13:53:05.102947 I | op-k8sutil: updating deployment rook-ceph-osd-1
2019-12-12 13:53:11.118711 I | op-k8sutil: finished waiting for updated deployment rook-ceph-osd-1
2019-12-12 13:53:11.118727 I | op-mon: this is not an upgrade, not performing upgrade checks
2019-12-12 13:53:11.118732 I | op-osd: started deployment for osd 1 (dir=false, type=bluestore)
2019-12-12 13:53:11.123851 I | op-osd: osd orchestration status for node worker-04 is computingDiff
2019-12-12 13:53:11.124080 I | op-osd: osd orchestration status for node worker-04 is orchestrating
2019-12-12 13:53:11.124134 I | op-osd: osd orchestration status for node worker-04 is completed
2019-12-12 13:53:11.124142 I | op-osd: starting 1 osd daemons on node worker-04
2019-12-12 13:53:11.131306 I | op-osd: deployment for osd 3 already exists. updating if needed
2019-12-12 13:53:11.133147 I | op-mon: this is not an upgrade, not performing upgrade checks
2019-12-12 13:53:11.133158 I | op-k8sutil: updating deployment rook-ceph-osd-3
2019-12-12 13:53:27.167921 I | op-k8sutil: finished waiting for updated deployment rook-ceph-osd-3
2019-12-12 13:53:27.167935 I | op-mon: this is not an upgrade, not performing upgrade checks
2019-12-12 13:53:27.167941 I | op-osd: started deployment for osd 3 (dir=false, type=bluestore)
2019-12-12 13:54:27.174051 I | op-osd: waiting on orchestration status update from 1 remaining nodes
2019-12-12 13:55:27.174172 I | op-osd: waiting on orchestration status update from 1 remaining nodes
2019-12-12 13:56:27.174297 I | op-osd: waiting on orchestration status update from 1 remaining nodes
W1212 13:56:33.453703       7 reflector.go:289] github.com/rook/rook/pkg/operator/ceph/cluster/controller.go:174: watch of *v1.ConfigMap ended with: too old resource version: 32130602 (32134373)
2019-12-12 13:57:27.174437 I | op-osd: waiting on orchestration status update from 1 remaining nodes
2019-12-12 13:58:27.174580 I | op-osd: waiting on orchestration status update from 1 remaining nodes
2019-12-12 13:59:27.174725 I | op-osd: waiting on orchestration status update from 1 remaining nodes
2019-12-12 14:00:27.174863 I | op-osd: waiting on orchestration status update from 1 remaining nodes
2019-12-12 14:01:27.175004 I | op-osd: waiting on orchestration status update from 1 remaining nodes
2019-12-12 14:02:27.175144 I | op-osd: waiting on orchestration status update from 1 remaining nodes
2019-12-12 14:03:27.175275 I | op-osd: waiting on orchestration status update from 1 remaining nodes
2019-12-12 14:04:27.175413 I | op-osd: waiting on orchestration status update from 1 remaining nodes
W1212 14:04:40.456959       7 reflector.go:289] github.com/rook/rook/pkg/operator/ceph/cluster/controller.go:174: watch of *v1.ConfigMap ended with: too old resource version: 32134581 (32138414)
2019-12-12 14:05:27.175545 I | op-osd: waiting on orchestration status update from 1 remaining nodes
2019-12-12 14:06:27.175677 I | op-osd: waiting on orchestration status update from 1 remaining nodes
2019-12-12 14:07:27.176110 I | op-osd: waiting on orchestration status update from 1 remaining nodes
2019-12-12 14:08:27.176241 I | op-osd: waiting on orchestration status update from 1 remaining nodes
2019-12-12 14:09:27.176372 I | op-osd: waiting on orchestration status update from 1 remaining nodes
2019-12-12 14:10:27.176507 I | op-osd: waiting on orchestration status update from 1 remaining nodes
2019-12-12 14:11:27.176627 I | op-osd: waiting on orchestration status update from 1 remaining nodes
W1212 14:11:55.460156       7 reflector.go:289] github.com/rook/rook/pkg/operator/ceph/cluster/controller.go:174: watch of *v1.ConfigMap ended with: too old resource version: 32138621 (32142065)
2019-12-12 14:12:27.176746 I | op-osd: waiting on orchestration status update from 1 remaining nodes
2019-12-12 14:13:27.176895 E | op-osd: timed out waiting for 1 nodes: &{values:map[worker-01:{}]}
2019-12-12 14:13:27.182646 I | op-osd: checking if any nodes were removed
2019-12-12 14:13:27.190867 I | op-osd: processing 0 removed nodes
2019-12-12 14:13:27.190878 I | op-osd: done processing removed nodes

```

* Cluster CR (custom resource), typically called `cluster.yaml`, if necessary
* Operator's logs, if necessary
* Crashing pod(s) logs, if necessary

 To get logs, use `kubectl -n <namespace> logs <pod name>`
When pasting logs, always surround them with backticks or use the `insert code` button from the Github UI.
Read [Github documentation if you need help](https://help.github.com/en/articles/creating-and-highlighting-code-blocks).

**Environment**:
* OS (e.g. from /etc/os-release):
* Kernel (e.g. `uname -a`):
* Cloud provider or hardware configuration:
* Rook version (use `rook version` inside of a Rook Pod):
* Storage backend version (e.g. for ceph do `ceph -v`):
* Kubernetes version (use `kubectl version`):
* Kubernetes cluster type (e.g. Tectonic, GKE, OpenShift):
* Storage backend status (e.g. for Ceph use `ceph health` in the [Rook Ceph toolbox](https://rook.io/docs/rook/master/ceph-toolbox.html)):

**Comment by leseb** [comment]
@travisn I believe you fixed this recently and it's coming in 1.2, right?

**Comment by travisn** [comment]
Yes actually the fix will also be coming in v1.1.8 soon...
